cli/cli-client/src/main/java/org/infinispan/cli/commands/User.java

```diff
@@ -0,0 +1,271 @@
+         invocation.getShell().writeln(userTool.describeUser(username));
+         return CommandResult.SUCCESS;
+      }
+   }
+
+   @CommandDefinition(name = Remove.CMD, description = "Removes a user", aliases = "rm")
+   public static class Remove extends CliCommand {
+      public static final String CMD = "remove";
+
+      @Argument(description = "The username for the user", required = true)
+      String username;
+
+      @Option(description = "The path of the users.properties file", name = "users-file", shortName = 'f')
+      String usersFile;
+
+      @Option(description = "The path of the groups.properties file", name = "groups-file", shortName = 'w')
+      String groupsFile;
+
+      @Option(description = "The server root", defaultValue = "server", name = "server-root", shortName= 's')
+      String serverRoot;
+
+      @Option(shortName = 'h', hasValue = false)
+      protected boolean help;
+
+      @Override
+      public boolean isHelp() {
+         return help;
+      }
+
+      @Override
+      public CommandResult exec(ContextAwareCommandInvocation invocation) throws CommandException {
+         UserTool userTool = new UserTool(serverRoot, usersFile, groupsFile);
+         userTool.removeUser(username);
+         return CommandResult.SUCCESS;
+      }
+   }
+
+   @CommandDefinition(name = Modify.CMD, description = "Modifies a user")
+   public static class Modify extends CliCommand {
+      public static final String CMD = "modify";
+
+      @Argument(description = "The username for the user", required = true)
+      String username;
+
+      @Option(description = "The password for the user", shortName = 'p')
+      String password;
+
+      @Option(description = "The realm ", defaultValue = UserTool.DEFAULT_REALM_NAME, shortName = 'r')
+      String realm;
+
+      @OptionList(description = "The algorithms used to encrypt the password", shortName = 'a')
+      List<String> algorithms;
+
+      @OptionList(description = "The groups the user should belong to", shortName = 'g')
+      List<String> groups;
+
+      @Option(description = "Whether the password should be stored in plain text", name = "plain-text")
+      boolean plainText;
+
+      @Option(description = "The path of the users.properties file", name = "users-file", shortName = 'f')
+      String usersFile;
+
+      @Option(description = "The path of the groups.properties file", name = "groups-file", shortName = 'w')
+      String groupsFile;
+
+      @Option(description = "The server root", defaultValue = "server", name = "server-root", shortName= 's')
+      String serverRoot;
+
+      @Option(shortName = 'h', hasValue = false)
+      protected boolean help;
+
+      @Override
+      public boolean isHelp() {
+         return help;
+      }
+
+      @Override
+      public CommandResult exec(ContextAwareCommandInvocation invocation) throws CommandException {
+         UserTool userTool = new UserTool(serverRoot, usersFile, groupsFile);
```

Comment:
can it fail if password and groups are `null`? (so, no change)

Reply:
Splitting the command makes this moot